### PR TITLE
RECIPE-58

### PR DESCRIPTION
**Pull request by PatsiChang**
Added Unit Test for Recipe Service
Link: https://smart-home-cloud.atlassian.net/browse/RECIPE-58

### CODE REVIEW

File: src/main/java/com/patsi/utils/ListHelper.java [hunk 1]
```diff
@@ -1,10 +1,24 @@
...
 import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
 
 public class ListHelper {
     public static <T> List<T> newList() {
         return new ArrayList<T>();
     }
+    public static <T> List<T> newListWithParam(T... elements) {
```
Comment (ChakShingLo) [inline src/main/java/com/patsi/utils/ListHelper.java:14]: Why need this helper? Why don't simply use List.of(...elements)?

File: src/test/java/RecipeServiceTest.java [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+            .thenReturn(validRecipeList);
+        when(recipeRepository.save(any()))
+            .thenReturn(validRecipe);
+    }
+
+    //getRecipe()
+    @Test
+    void testGetRecipeWithValidUid() {
+        assertEquals(validRecipeList, recipeService.getRecipe(validUid));
+        verify(recipeRepository).findByUid(validUid);
```
Comment (ChakShingLo) [inline src/test/java/RecipeServiceTest.java]: Is `verify(recipeRepository).findByUid(validUid);` really necessary  in these test? 
asserting correct recipeList are returned should be sufficient.

Comment (PatsiChang) [reply src/test/java/RecipeServiceTest.java]: you're right :0

File: src/main/java/com/patsi/utils/ListHelper.java (hunk 1 above)
Comment (PatsiChang) [reply src/main/java/com/patsi/utils/ListHelper.java:14]: oh yes I fixed it now :D

**Review by ChakShingLo — APPROVED** [review]

**Pull request merged**